jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/ProjectCommand.java

```diff
@@ -94,24 +94,25 @@ private static class Classpath {
     public static List<PackageNode> listProjects(List<Object> arguments, IProgressMonitor monitor) {
         String workspaceUri = (String) arguments.get(0);
         IPath workspaceFolderPath = ResourceUtils.canonicalFilePathFromURI(workspaceUri);
-        String invisibleProjectName = ProjectUtils.getWorkspaceInvisibleProjectName(workspaceFolderPath);
 
-        IProject[] projects = getWorkspaceRoot().getProjects();
+        IJavaProject[] javaProjects = ProjectUtils.getJavaProjects();
         ArrayList<PackageNode> children = new ArrayList<>();
-        List<IPath> paths = Collections.singletonList(workspaceFolderPath);
-        for (IProject project : projects) {
-            if (!project.isAccessible() || !ProjectUtils.isJavaProject(project)) {
+        for (IJavaProject javaProject : javaProjects) {
+            IProject project = javaProject.getProject();
+            if (!project.isAccessible() || project.getLocation() == null) {
                 continue;
             }
-            // Ignore all the projects that's not contained in the workspace folder, except
-            // for the invisible project.
-            // This check is needed in multi-root scenario.
-            if ((!ResourceUtils.isContainedIn(project.getLocation(), paths)
-                    && !Objects.equals(project.getName(), invisibleProjectName))) {
+
+            // ignore default projects
+            if (Objects.equals(project.getName(), ProjectsManager.DEFAULT_PROJECT_NAME)) {
                 continue;
             }
+
             PackageNode projectNode = PackageNode.createNodeForProject(JavaCore.create(project));
-            if (Objects.equals(project.getName(), invisibleProjectName)) {
+
+            // set the folder name as the project name when the project location
+            // is out of the workspace folder.
+            if (!workspaceFolderPath.isPrefixOf(project.getLocation())) {
```

Comment:
is it possible that there are multiple projects out of the workspace folder? In this scenario, they will have the same name.

Reply:
Yes, that could be, though not very likely to happen. We can consider showing the path of the node in the tooltip, like the behavior in file explorer, if we see users complaining about this.